lnpbp-0001.md

```diff
@@ -125,43 +125,43 @@ algorithm, that avoids all of those issues.
+runs as follows:
 
-1. Reduce set `P` to `P*` by removing all duplicated public keys.
-2. Compute a sum `S` of all unique public keys in set `P*`; fail the protocol if
+1. Reduce set `P` to `P*` by removing all duplicate public keys.
+2. Compute sum `S` of all unique public keys in set `P*`; fail the protocol if
    an overflow over elliptic curve generator point order happens during the 
    procedure.
 3. Construct a byte string `lnpbp1_msg`, composed of the original message
    prefixed with a single SHA256 hash of `LNPBP1` string and a single SHA256
-   hash of protocol-specific tag:  
+   hash of the protocol-specific `tag`:  
    `lnpbp1_msg = SHA256("LNPBP1") || SHA256(tag) || msg`
-4. Compute HMAC-SHA256 of the `lnbp1_msg` using sum of public keys `S`. The 
+4. Compute HMAC-SHA256 of `lnbp1_msg` using the sum of public keys `S`. The 
    resulting value is named **tweaking factor** `f`:  
    `f = HMAC-SHA256(lnpbp1_msg, S)`
-5. Make sure that the tweaking factor is less than order `n` of a generator 
+5. Make sure that the tweaking factor is less than the order `n` of a generator 
    point of the used elliptic curve, such that no overflow can happen when it is 
    added to the original public key. If the order is exceeded, fail the protocol
    indicating the reason of failure.
-6. Multiply the tweaking factor on the used elliptic curve generator point `G`:  
+6. Multiply the tweaking factor `f` on the used elliptic curve generator point `G`:  
    `F = G * f`
-7. Check that the result of 5 is not equal to the point-at-infinity; otherwise 
+7. Check that the result of 6 is not equal to the point-at-infinity; otherwise 
    fail the protocol, indicating the reason of failure, such that the protocol 
-   may be run with another initial public key `P'` value.
-8. Add two elliptic curve points: the original public key `Po` and 
-   point `F`, derived from the tweaking-factor. This will give resulting tweaked 
+   may be run with another initial public key set `P'`.
+8. Add the two elliptic curve points: the original public key `Po` and the
+   point `F`, derived from the tweaking-factor. This will result in a tweaked 
    public key `T`: `T = Po + F`. Check that the result is not equal to the 
-   point-at-infinity of the elliptic curve and fail the protocol otherwise, 
+   point-at-infinity of the elliptic curve or fail the protocol otherwise, 
    indicating the reason of failure, such that the protocol may be run with 
-   another initial public key set.
+   another initial public key set `P'`.
 
 The final formula for the commitment is:  
 `T = Po + G * HMAC-SHA256(SHA256("LNPBP1") || SHA256(tag) || msg, S)`
 
 ### Verification procedure
 
 **Verification procedure** for the commitment (i.e. tweaked public key `T`) can 
-be performed with the provision of the original public key `P` and message `msg` 
+be performed with the provision of the original public key `Po` and message `msg` 
```

Comment:
What about the sum `S` of the public key set `P`or `P*`? Isn't that required as well for the verification procedure?

Reply:
Yes, we need to add it. Original LNPBP-1 draft was written for a single public key, but later was updated to match LNPBP-2 script situation, and this part was missed. LNPBP-2 requires the full script with all public keys to be present, however we need to mention this requirement here as well (probably at LNPBP-1 we should not talk about the script but rather about set of keys `P*`)